connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -0,0 +1,180 @@
+import { ConnectorResourceType, ModelId } from "@dust-tt/types";
+import { hash as blake3 } from "blake3";
+import { CheerioCrawler, Configuration } from "crawlee";
+import turndown from "turndown";
+
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertToDatasource } from "@connectors/lib/data_sources";
+import { Connector } from "@connectors/lib/models";
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+  WebCrawlerPage,
+} from "@connectors/lib/models/webcrawler";
+import {
+  reportInitialSyncProgress,
+  syncSucceeded,
+} from "@connectors/lib/sync_status";
+
+export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found.`);
+  }
+  const webCrawlerConfig = await WebCrawlerConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!webCrawlerConfig) {
+    throw new Error(`Webcrawler configuration not found for connector.`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  let pageCount = 0;
+  let errorCount = 0;
+  const createdFolders = new Set<string>();
+
+  const crawler = new CheerioCrawler(
+    {
+      maxRequestsPerCrawl: 300,
+      maxConcurrency: 10,
+
+      async requestHandler({ $, request, enqueueLinks }) {
+        const extracted = new turndown()
+          .remove(["style", "script", "iframe"])
+          .turndown($.html());
+
+        const pageTitle = $("title").text();
+
+        await enqueueLinks({
+          transformRequestFunction: (req) =>
+            req.url.startsWith(webCrawlerConfig.url) ? req : false,
```

Comment:
What if I use `https://test.com/superinfo/index.html`  as input?

Reply:
then we will only crawl urls that are the childs of `https://test.com/superinfo/index.html`.

The UI specifies that we'll only take childrens of this URL.